### ISSUE TITLE
Problem extracting data (Fields related)

### ISSUE THREAD

**Issue by marceloyb** [diogofgm/kaspersky_app#2]
Hi, i downloaded the app and started using it. I had to make some adaptations on the search of the dashboards, since in my Splunk the fields you were searching are not default extracted. The thing is, in some dashboards i managed to do fine in the extracting, but in some others i can't get trustable information. Could you please share how you did your field extraction?

**Comment by diogofgm** [comment]
Hi Marcelo 
Have you installed the add on? Also what format are you using? CEF or LEEF?

**Comment by marceloyb** [comment]
Yes, addon is installed. I believe CEF, since it's been configured directly in KSC Panel.

![image](https://user-images.githubusercontent.com/20404114/66764623-8d93a980-ee80-11e9-95f8-a53c48a1a8d0.png)

**Comment by diogofgm** [comment]
You should be using LEEF. If I recall correctly you need to choose QRadar in the format to have it output LEEF. Splunk option wasn’t available when I started building these add nos and app.

**Comment by marceloyb** [comment]
Tried changing to LEEF (QRadar) and Splunk stopped receiving the logs if i use sourcetype kaspersky:leef. Any ideas why? If i use the "syslog" sourcetype it works.

Edit: Actually it is receiving, but the kaspersky:leef sourcetype is somehow throwing events into my indexer with wrong time stamp (4 hours in the future) 
![image](https://user-images.githubusercontent.com/20404114/66774636-647f1300-ee98-11e9-9ddf-6691ea122a7a.png)

**Comment by diogofgm** [comment]
can you post a sanitised sample of you data? the only reason to have data generated with a timestamp in the future would be if you have problems with timezone or misconfigured server times since the add on is not doing anything regarding time. Check the time of you splunk server and kaspersky server. Are you using a common ntp server?

**Comment by diogofgm** [comment]
After looking at you profile I saw you might be in GMT-3. I was thinking, if I'm correct 4h in the future would be something like your data is coming in as UTC if you are under DST. 
Thats something it never occurred to me and I actually didn't configured in the TA since in Portugal we are GMT/UTC (Apart from when we are GMT+1 due to DST). Its possible that kaspersky is outputting data in UTC and since no TZ is defined you get data in the future.

**Comment by marceloyb** [comment]
Probably. The problem happens once the data get into the TA i made a change on the sourcetype kaspersky:leef so the timestamp will be defined on the Current time, but there are still some discrepancies. Since i'm kinda newbie on Splunk, i couldn't find where i could fix the cause, but with the changes the app is working fine.
![image](https://user-images.githubusercontent.com/20404114/66828490-85427980-ef27-11e9-8d66-4f57f6c17d0b.png)

**Comment by diogofgm** [comment]
Have you compared the timestamp and TZ of the server where you are hosting the Kaspersky SC console with the on in the logs? 
What is the TZ you have setup in your splunk profile (Your account name > preferences)?
In the image you posted, what is the right timestamp? the one in splunk or the one in the logs?

The thing here is, when you index data you can either extract or assign a TZ to your data. This allows splunk to adjust it to TZ the one you have in your profile.

From splunk docs:
https://docs.splunk.com/Documentation/Splunk/7.3.2/Admin/Propsconf
```
TZ = <timezone identifier>
* The algorithm for determining the time zone for a particular event is as
  follows:
  * If the event has a timezone in its raw text (for example, UTC, -08:00),
  use that.
  * If TZ is set to a valid timezone string, use that.
  * If the event was forwarded, and the forwarder-indexer connection uses
  the version 6.0 and higher forwarding protocol, use the timezone provided
  by the forwarder.
  * Otherwise, use the timezone of the system that is running splunkd.
* Default: empty string
```
